### PR TITLE
adding Bt to cancel wait, spin and backup

### PR DESCRIPTION
**Pull request by padhupradheep**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   |  |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Neobotix robots |

---

## Description of contribution in a few bullet points

* Added Cancel Feature for WaitAction, SpinAction

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [x] Check that any new parameters added are updated in navigation.ros.org
- [x] Check that any significant change is added to the migration guide
- [x] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [x] Check that any new functions have Doxygen added
- [x] Check that any new features have test coverage
- [x] Check that any new plugins is added to the plugins page
- [x] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by padhupradheep** [comment]
Will remove the draft, once I'm done testing it on an actual BT.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Looks good otherwise!

**Comment by SteveMacenski** [comment]
If you go to the tests, you'll see about 2-3 dozen linting errors due to changes in formatting. If you could manually pick off 5-10 of them, I'd appreciate the incremental help

File: nav2_behavior_tree/nav2_tree_nodes.xml
Comment (SteveMacenski) [inline nav2_behavior_tree/nav2_tree_nodes.xml:21]: Is there also not `server_timeout` as your PR in documentation shows?

@padhupradheep please, please slow down and think through these requested changes. Don't just blitz through them. For every 10 of these easily overlookable errors I find, there could easily be 1-2 that slip through. I appreciate your help in Nav2, but please go slow, be methodical, make sure things are complete and correct.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_behavior_tree/nav2_tree_nodes.xml
Comment (SteveMacenski) [reply nav2_behavior_tree/nav2_tree_nodes.xml:21]: Its better to get 1 thing done, and done well, than 5 things sent but then we spend 2 months on getting it to be merged :+1: :wink:

Comment (padhupradheep) [reply nav2_behavior_tree/nav2_tree_nodes.xml:21]: I agree on the point. I actually missed it in the cancelcontrol BT as well. That got me a bit off guard. Sorry for the trouble.

Comment (SteveMacenski) [reply nav2_behavior_tree/nav2_tree_nodes.xml:21]: No worries, not trying to scold, just letting you know I think you can do better :wink: Attention to detail is the game of engineering. I was just like you for a number of years until I started leading projects and having my own hastfulness bite me in the butt. It has made me more cautious and thoughtful - I think you'll find that too over time as you work on the same systems year after year. Old decisions you made come back to haunt you :laughing:

Comment (padhupradheep) [reply nav2_behavior_tree/nav2_tree_nodes.xml:21]: No issues, I really understand that. I’m happy to get some honest feedback on the fly from you, which really helps me develop as a engineer. But thanks for being patient. Let me see what I missed and add it ✌️

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Waiting on CI before merge

**Pull request merged**
